### PR TITLE
Add csharp docfx

### PR DESCRIPTION
**Pull request by cassiebreviu**
**Description**: Add ApiDocs project to implement DocFX for C# automatic documentation.

Staged changes: https://cassiebreviu.github.io/onnxruntime/docs/api/csharp/dest/api/index.html

**Motivation and Context**
- Better and automatic documentation for csharp onnxruntime API.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/win-cpu-ci.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
   workspace:
```
[the remark this answers is not in the document]
Comment (cassiebreviu) [reply tools/ci_build/github/azure-pipelines/templates/win-cpu-ci.yml:47]: I removed my changes so correct, all thats left is white space changes.

**Comment by yuslepukhin** [comment]
> {"message":"Invalid triple slash comment is ignored: \r\n","source":"MetadataCommand.ExtractMetadata","date_time":"2022-08-05T20:37:32.4351086Z","message_severity":"warning","correlation_id":"10469BBD-CD6E-41B5-946A-A12433ADF511.1.1.9"}

This seems to have lots of errors. Do we want to add this to the PR?

---
Refers to: csharp/ApiDocs/log.txt:3 in 7c4b04b. [](commit_id = 7c4b04b09b1178f6019cb37b5475219c21569b96, deletion_comment = False)

**Comment by cassiebreviu** [comment]
> > {"message":"Invalid triple slash comment is ignored: \r\n","source":"MetadataCommand.ExtractMetadata","date_time":"2022-08-05T20:37:32.4351086Z","message_severity":"warning","correlation_id":"10469BBD-CD6E-41B5-946A-A12433ADF511.1.1.9"}
> 
> This seems to be a build log with lots of errors. Do we want to add this to the PR?
> 
> Refers to: csharp/ApiDocs/log.txt:3 in [7c4b04b](https://github.com/microsoft/onnxruntime/commit/7c4b04b09b1178f6019cb37b5475219c21569b96). [](commit_id = [7c4b04b](https://github.com/microsoft/onnxruntime/commit/7c4b04b09b1178f6019cb37b5475219c21569b96), deletion_comment = False)

These are just warnings. They are issues with how a comment is formatted in the source code. It is something that could be fixed at a later date.

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Review by pranavsharma — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/win-cpu-ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/win-cpu-ci.yml:47]: Then I think it's better to not having this file in your PR. What do you think?

**Review by snnn — APPROVED** [review]
I only reviewed one file: win-cpu-ci.yml. 
But the changes are unnecessary. If you didn't include it in this PR, you don't need to wait an approval from me. 
To save your time: I approved it.

**Pull request merged**